### PR TITLE
Update versions in pom, specifically project to 3.0/EE 10

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

### CODE REVIEW

**Comment by pnicolucci** [comment]
Hey @arjantijms can you explain why you're proposing removing the java 11 enforcement? I had done that as part of this [issue](https://github.com/eclipse-ee4j/jstl-api/issues/156 ). Did I go too far with the restriction?

**Comment by arjantijms** [comment]
> Hey @arjantijms can you explain why you're proposing removing the java 11 enforcement?

There should be no restriction removed. JDK 11 is already set for the compiler, so there should be no additional need to specify it in the enforcer. Just trying to cut the duplication and ceremony a little ;)

File: spec/pom.xml [hunk 1]
```diff
@@ -62,11 +61,11 @@
     <build>
         <defaultGoal>package</defaultGoal>
         <plugins>
-            <!-- Sets minimal Maven version to 3.5.4 and Java version to 11 -->
+            <!-- Sets minimal Maven version to 3.6.0 and Java version to 11 -->
```
Comment (pnicolucci) [inline spec/pom.xml:64]: Since we're removing the java 11 aspects of this plugin let's update the comment as was done in the other poms.

Comment (arjantijms) [reply spec/pom.xml]: Okay, it is done.

**Review by volosied — COMMENTED** [review]
I tested locally, and this look good overall. (Although I needed some prep work to build the latest el, servlet, and jsp apis first). 

It also looks like our previous `mvn javadoc:javadoc` errors went away. 

However, I spotted one issue when building. It looks like there an extra apidocs folder created under the api directory? I'm not sure why? 
```
[WARNING] Javadoc Warnings
[WARNING] javadoc: warning - The code being documented uses packages in the unnamed module, but the packages defined in https://projects.eclipse.org/projects/ee4j.jstl/apidocs/ are in named modules.

[INFO] --- maven-javadoc-plugin:3.3.1:jar (attach-javadocs) @ jakarta.servlet.jsp.jstl ---
[ERROR] Error fetching link: jstl-api/api/apidocs. Ignored it
```

**Comment by arjantijms** [comment]
> Although I needed some prep work to build the latest el, servlet, and jsp apis first).

Often the easiest thing is to use staging, e.g. add -Pstaging when building. Expression Language and others are available in the staging repo.

**Comment by arjantijms** [comment]
> It looks like there an extra apidocs folder created under the api directory? I'm not sure why?

In general there's "weird" errors as javadoc is transitioning to being module aware. I've tried to solve a few of them, but almost always they are unsolvable at the project level. If you find a solution for this particular one though it would be good to add.

**Review by pnicolucci — APPROVED** [review]
Looks good. Thanks!

**Review by volosied — APPROVED** [review]
I couldn't figure the apidoc folder issue, but I don't think it should hold us back.  Approved as well.

**Comment by arjantijms** [comment]
Thanks!

**Pull request merged**